boards/posix/native_posix/hw_models_top.c

```diff
@@ -92,12 +98,17 @@ static void hwm_sleep_until_next_timer(void)
 		/* LCOV_EXCL_STOP */
 	}
 
-	if (signaled_end || (simu_time > end_of_time)) {
+	_SAFE_CALL(pthread_mutex_lock(&exit_mutex));
```

Comment:
Technically what you do is correct: you lock a mutex to protect access to the shared variables `hwm_exit_request` and `hwm_exit_code`.
But this loop executes for each simulation "delta cycle". Locking and unlocking a mutex in each iteration adds quite a bit of ovehead.
So, instead of doing it this way, I would:
* remove `hwm_exit_request` and have `hwm_main_loop_exit` set `signaled_end`.
* Have the mutex here protect only the access to hwm_exit_code (so move the mutex lock call inside the if)

Why: signaled_end starts as 0, and is only set to something != 0 by another thread when it wants things to end. It is safe for the hwm thread to read that variable at any moment without protection, because in the worst case it just waits for one hw model cycle more before stopping. Here, there is no race we need to protect against. 
(see other comments for more thoughts)